src/dodal/devices/stepped_grid_scan.py

```diff
@@ -0,0 +1,285 @@
+import threading
+import time
+from dataclasses import dataclass
+
+from bluesky.plan_stubs import mv
+from dataclasses_json import DataClassJsonMixin
+from ophyd import (
+    Component,
+    Device,
+    EpicsSignal,
+    EpicsSignalRO,
+    EpicsSignalWithRBV,
+    Signal,
+)
+from ophyd.status import DeviceStatus, StatusBase
+
+from dodal.devices.motors import XYZLimitBundle
+from dodal.devices.status import await_value
+from dodal.parameters.experiment_parameter_base import AbstractExperimentParameterBase
+from dodal.utils import Point3D
+
+
+@dataclass
+class GridAxis:
+    start: float
+    step_size: float
+    full_steps: int
+
+    def steps_to_motor_position(self, steps):
+        return self.start + (steps * self.step_size)
+
+    @property
+    def end(self):
+        return self.steps_to_motor_position(self.full_steps)
+
+    def is_within(self, steps):
+        return 0 <= steps <= self.full_steps
+
+
+@dataclass
+class SteppedGridScanParams(DataClassJsonMixin, AbstractExperimentParameterBase):
```

Comment:
Should: I think it makes sense for these parameters to be on the dodal side for the fast grid scan as they directly relate to the hardware but for the stepped scan the parameters should probably be on the artemis side (see https://github.com/DiamondLightSource/python-artemis/blob/main/src/artemis/parameters/internal_parameters/plan_specific/rotation_scan_internal_params.py)